sample_consensus/include/pcl/sample_consensus/impl/sac_model_cylinder.hpp

```diff
@@ -55,16 +55,28 @@ pcl::SampleConsensusModelCylinder<PointT, PointNT>::isSampleGood (const Indices
     PCL_ERROR ("[pcl::SampleConsensusModelCylinder::isSampleGood] Wrong number of samples (is %lu, should be %lu)!\n", samples.size (), sample_size_);
     return (false);
   }
-  return (true);
+
+  // Make sure that the two sample points are not identical
+  if (
+      std::abs ((*input_)[samples[0]].x - (*input_)[samples[1]].x) > std::numeric_limits<float>::epsilon ()
+    ||
+      std::abs ((*input_)[samples[0]].y - (*input_)[samples[1]].y) > std::numeric_limits<float>::epsilon ()
+    ||
+      std::abs ((*input_)[samples[0]].z - (*input_)[samples[1]].z) > std::numeric_limits<float>::epsilon ())
+  {
+    return (true);
+  }
+
+  return (false);
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 template <typename PointT, typename PointNT> bool
 pcl::SampleConsensusModelCylinder<PointT, PointNT>::computeModelCoefficients (
       const Indices &samples, Eigen::VectorXf &model_coefficients) const
 {
-  // Need 2 samples
-  if (samples.size () != sample_size_)
+  // Make sure that the samples are valid
+  if (!isSampleGood (samples))
   {
     PCL_ERROR ("[pcl::SampleConsensusModelCylinder::computeModelCoefficients] Invalid set of samples given (%lu)!\n", samples.size ());
```

Comment:
Sure.

Am I correct in my interpretation of the contribution guidelines that it's preferred to have a clean commit history, i.e. squash your recommended changes into the existing commits?

Reply:
During the reviewing process it is usually clearer when new changes are added as new commits. When we are all happy with the changes, we can think about cleaning up the commit history by squashing some commits, or squash _all_ commits into a single commit when merging

Reply:
What do you think of "inverting" the similarity condition so that control flow of `isSampleGood` is more "symmetric"? Now (or soon to be):

```cpp
template <typename PointT> bool
pcl::SampleConsensusModelLine<PointT>::isSampleGood (const Indices &samples) const
{
  if (samples.size () != sample_size_)
  {
    PCL_ERROR ("[pcl::SampleConsensusModelLine::isSampleGood] Wrong number of samples (is %lu, should be %lu)!\n", samples.size (), sample_size_);
    return (false);
  }

  // Make sure that the two sample points are not identical
  if (
      std::abs ((*input_)[samples[0]].x - (*input_)[samples[1]].x) > std::numeric_limits<float>::epsilon ()
    ||
      std::abs ((*input_)[samples[0]].y - (*input_)[samples[1]].y) > std::numeric_limits<float>::epsilon ()
    ||
      std::abs ((*input_)[samples[0]].z - (*input_)[samples[1]].z) > std::numeric_limits<float>::epsilon ())
  {
    return (true);
  }

  PCL_ERROR ("[pcl::SampleConsensusModelLine::isSampleGood] The two sample points are too similar!\n");
  return (false);
}
```

vs.

```cpp
template <typename PointT> bool
pcl::SampleConsensusModelLine<PointT>::isSampleGood (const Indices &samples) const
{
  if (samples.size () != sample_size_)
  {
    PCL_ERROR ("[pcl::SampleConsensusModelLine::isSampleGood] Wrong number of samples (is %lu, should be %lu)!\n", samples.size (), sample_size_);
    return (false);
  }

  // Make sure that the two sample points are not identical
  if (
      std::abs ((*input_)[samples[0]].x - (*input_)[samples[1]].x) <= std::numeric_limits<float>::epsilon ()
    &&
      std::abs ((*input_)[samples[0]].y - (*input_)[samples[1]].y) <= std::numeric_limits<float>::epsilon ()
    &&
      std::abs ((*input_)[samples[0]].z - (*input_)[samples[1]].z) <= std::numeric_limits<float>::epsilon ())
  {
    PCL_ERROR ("[pcl::SampleConsensusModelLine::isSampleGood] The two sample points are too similar!\n");
    return (false);
  }

  return (true);
}
```

Reply:
Sounds good!

Reply:
A side effect of this is that the test `SampleConsensusModelLine.SampleValidationPointsEqual` now got a verbose even without the manually increased verbosity level since in the test case this branch is hit every time until the iteration count is exhausted (1000x if i'm not mistaken).